src/librbd/AioRequest.cc

```diff
@@ -187,7 +288,34 @@ namespace librbd {
 				 << m_object_image_extents << dendl;
 
 	  m_state = LIBRBD_AIO_WRITE_COPYUP;
-	  read_from_parent(m_object_image_extents);
+
+          if (is_copy_on_read(m_ictx, m_snap_id)) {
+	    m_ictx->copyup_list_lock.Lock();
```

Comment:
Indeed, read_from_parent can delete CopyupRequest, which triggers copyup_list manipulation. But within the deconstructor of CopyupRequest, copyup_list will be protected explicitly. Thus if we use Mutex::Locker, it will result in deadlock.